js/NavMainComponent.js

```diff
@@ -2,104 +2,249 @@
 
-    component.$html.find('.nav-secondary .nav-list').removeClass('is-active');
+    component.$navSecondary.find('.nav-list').removeClass('is-active');
+
+    if (component.$navQuaternary.hasClass('is-open')) {
+        component.$navQuaternary.removeClass('is-open');
+        component.$navQuaternary.find('.nav-list.is-active').removeClass('is-active');
+    } else {
+        component.$navTertiary.removeClass('is-open');
+    }
+};
+
+// Detect window height, adjust the number of items in the primary nav and check when to add "More" option
+NavMainComponent.prototype.adjustNavItems = function () {
+    var component = this,
+        availableHeight = component.$window.height(),
+        navItemsHeight = (component.$html.find('.nav-primary .nav-items').outerHeight(true) + component.$html.find('.jadu-branding').outerHeight(true)),
+        moreIconHeight = 72, // Pre calculated height of the "More" nav item
+        navItemsCountTotal = component.$html.find('.nav-primary .nav-items li').length,
+        numberOfHiddenNavItems = 0;
+
+    if (navItemsHeight + moreIconHeight > availableHeight) {
+        // If there is not enough space hide the last primary nav items
+        component.hidePrimaryNavItems(navItemsHeight, moreIconHeight, availableHeight);
+        // Get the number of hidden items to make only them visible in the tertiary menu
+        numberOfHiddenNavItems = component.$html.find('.nav-primary .nav-items li:hidden').length;
+        // Add "More" nav item and check its visibility if already exists
+        component.addMoreNavItem(numberOfHiddenNavItems);
+        // Hide the primary nav items duplicate in tertiary menu
+        component.hideMoreCategoriesTopItems(navItemsCountTotal, numberOfHiddenNavItems);
+    } else {
+        // Unhide items if they were hidden and there is space in the primary nav
+        component.$navPrimary.find('.nav-items li').show();
+        component.$navPrimary.find('.nav-items [label="More"]').hide();
+    }
+};
+
+NavMainComponent.prototype.hidePrimaryNavItems = function (navItemsHeight, moreIconHeight, availableHeight) {
+    var component = this,
+        lastPrimaryNavItem = component.$html.find('.nav-primary .nav-items li:last-child'),
+        nthChild = 2; // This number represents the item before the last in the nth-last-child;
+
+    while (navItemsHeight + moreIconHeight > availableHeight) {
+        // If last nav item is visible hide it
+        if (lastPrimaryNavItem.is(':visible')) {
+            lastPrimaryNavItem.hide();
+        } else {
+            // If last nav item is hidden hide the next one up
+            component.$html.find('.nav-primary .nav-items li:nth-last-child('+ nthChild +')').hide();
+            nthChild++;
+        }
+
+        // Recalculate nav items height based on items just hidden
+        navItemsHeight = (component.$html.find('.nav-primary .nav-items').outerHeight(true) + component.$html.find('.jadu-branding').outerHeight(true));
```

Comment:
Tried to refactor but I get undefined while using the '.outerHeight(true)' @mikedevelops